### PR TITLE
Sysrst testplan update

### PR DESCRIPTION
**Pull request by pamaury**

### CODE REVIEW

File: hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson [hunk 1]
```diff
@@ -18,6 +18,10 @@
       si_stage: SV3
       lc_states: ["DEV", "PROD", "PROD_END", "RMA"]
       tests: ["chip_sw_pwrmgr_sysrst_ctrl_reset", "chip_sw_sysrst_ctrl_reset"]
+      bazel: [
+        "//sw/device/tests:pwrmgr_deep_sleep_all_reset_reqs_test",
+        "//sw/device/tests:pwrmgr_normal_sleep_all_reset_reqs_test",
```
Comment (jwnrt) [inline hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson:23]: Are you sure these two tests definitely cover the test plan above?

It's not obvious from the C file that they do the steps listed in the description. Are these things happening behind the scenes?

Comment (pamaury) [reply hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson:23]: This is done in [the pwrmgr reset library](https://github.com/lowRISC/opentitan/blob/master/sw/device/tests/pwrmgr_sleep_resets_lib.c#L181). The function `config_sysrst` is called in both tests and configures the sysrst block to reset when a specific combo is detected.

**Review by a-will — APPROVED** [review]

File: hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson (hunk 1 above)
Comment (jwnrt) [reply hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson:23]: Aside, should that C file move to `//sw/device/lib/testing/...` with the other testutils, or is it different in some way?

Comment (pamaury) [reply hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson:23]: I think it's a little bit too specific to be in testutils. Basically there are four very similar tests that involve doing resets in mutiple different ways (pwrmgr, sysrst, alertmgr and wdog). This library is there to configure those 4 sources and factor out the code of those 4 tests. But pretty much everything in this library is specific to those tests.

I think it should be the other way around: those four tests should probably have only one C file that uses one or two defines to change the maybe one or two things that differ between them.

File: hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson [hunk 2]
```diff
@@ -18,6 +18,10 @@
       si_stage: SV3
       lc_states: ["DEV", "PROD", "PROD_END", "RMA"]
       tests: ["chip_sw_pwrmgr_sysrst_ctrl_reset", "chip_sw_sysrst_ctrl_reset"]
+      bazel: [
+        "//sw/device/tests:pwrmgr_deep_sleep_all_reset_reqs_test",
```
Comment (matutem) [inline hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson:22]: I noticed in pwrmgr_sleep_all_wake_ups_impl.c we just clear the wakeup_status CSR, but don't check it was set.

File: hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson [hunk 3]
```diff
@@ -76,6 +80,7 @@
         si_stage:SV3
         lc_states: ["DEV", "PROD", "PROD_END", "RMA"]
         tests: ["chip_sw_sysrst_ctrl_ulp_z3_wakeup"]
+        bazel: ["//sw/device/tests:pwrmgr_random_sleep_all_wake_ups"]
```
Comment (matutem) [inline hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson:83]: This test uses the regular KEY_INTR wakeup, not ULP, so it is not a good match to this testpoint.

By the way, in line 78 above there is a typo in ULATRA. I guess we don't check for legal features.

File: hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson (hunk 2 above)
Comment (pamaury) [reply hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson:22]: I had a look at this but it seems that pwrmgr_sleep_all_wake_ups_impl.c has `check_wakeup_reason` which is called and that checks the wakeup reason? I am missing something?

Also I noticed that in the sival test, the code sets up a combo with all keys:
https://github.com/lowRISC/opentitan/blob/master/sw/device/tests/pwrmgr_sleep_resets_lib.c#L176
but actually only configures the pinmux for Key0. Should the combo be set up for Key0 only?

File: hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson (hunk 3 above)
Comment (pamaury) [reply hw/top_earlgrey/data/ip/chip_sysrst_ctrl_testplan.hjson:83]: You are right, I will port the `sysrst_ctrl_ulp_z3_wakeup_test` test to sival instead.

Unfortunately we do not check for legal features but I believe that there was a plan to implement that.

**Pull request closed without merging**